frontend/src/datasources/UserDataSource.ts

```diff
@@ -53,4 +52,24 @@ export class UserDataSource {
         return { error: "" };
     }
 
+    public async getLeaguesListByUser(): Promise<
+    { error: string ; leagues ?  : {id: string, name: string}[]}
```

Comment:
tu pourrais faire de  {id: string, name: string} un type, comme tu as fait dans League.tsx avec un export (mais plutot a faire dans les datasources et tu import dans League.tsx)
```suggestion
    { error: string } | { leagues: {id: string, name: string}[] }
```